base/arithmetic.h

```diff
@@ -41,6 +34,22 @@ T euclidean_mod(T a, T b, T define_b_zero = 0) {
   return r >= 0 ? r : (b < 0 ? r - b : r + b);
 }
 
+template <typename T>
+T euclidean_div_positive_divisor(T a, T b) {
+  assert(b > 0);
+  T q = a / b;
+  T r = a % b;
+  T rs = r >> (sizeof(T) * 8 - 1);
```

Comment:
Sorry, I don't understand what is happening here, could you please add a comment?